addons/account/models/account_invoice.py

```diff
@@ -604,13 +620,77 @@ def action_invoice_sent(self):
             'context': ctx,
         }
 
+    def _invoice_email_split(self, msg):
+        return email_split((msg.get('from') or '') + ',' + (msg.get('cc') or ''))
+
     @api.multi
     @api.returns('self', lambda value: value.id)
     def message_post(self, **kwargs):
         if self.env.context.get('mark_invoice_as_sent'):
             self.filtered(lambda inv: not inv.sent).write({'sent': True})
         return super(AccountInvoice, self.with_context(mail_post_autofollow=True)).message_post(**kwargs)
 
+    @api.model
+    def message_new(self, msg_dict, custom_values=None):
+        # Split `From` and `CC` email address from received email and check if any of odoo partner is associated with that email
+        subscribe_partner_ids = partner_ids = [pid for pid in self._find_partner_from_emails(self._invoice_email_split(msg_dict), force_create=False) if pid]
+        # If sender of the email is the one from odoo internal user then it is likely that the vendor forwarded bill as an email to the internal user and inturn,
+        # internal user forwarded that email to vendor bill alias to automatically generate vendor bill on behalf of vendor. In this case, we will look for
+        # vendor's email address in email's body(generally it will be found first email address under forwarded Message section) and we will try to find vendor
+        # in odoo that is accociated with that email and will set that Vendor in the generated bill. If it can't be found, then it will left blank and someone
+        # has to manually create vendor based on chatter message.
+        user = self.env['res.users'].search([('partner_id', 'in', partner_ids)], limit=1)
+        if user:
+            partner_id = False
+            email_addresses = email_re.findall(msg_dict.get('body'))
+            if email_addresses:
+                partner_ids = [pid for pid in self._find_partner_from_emails([email_addresses[0]], force_create=False) if pid]
+                # this is an actual vendor
+                partner_id = partner_ids and partner_ids[0]
+        else:
+            # this is an actual vendor
+            partner_id = msg_dict.get('author_id')
+        if partner_id:
+            subscribe_partner_ids.append(partner_id)
+        # "TO" email address will generally be the vendor journal alias, and we will set that alias in origin field of vendor bill
+        # which was generated from email alias.
+        [to_email] = email_split((msg_dict.get('to') or ''))
+        alias_name = to_email.split('@')
+        [from_email] = email_split((msg_dict.get('from') or ''))
+        values = dict(custom_values or {}, partner_id=partner_id, source_email=from_email)
+        journal = self.env['account.journal'].search([
```

Comment:
It should be in the alias default values, not computed strangely here.

Reply:
@tde-banana-odoo I didn't get your comment, can you please come again on this ?
Actually what we did here is, for example, when a system is having two `purchase` type journals defined with it's dedicate aliases(say, Journal1 with vendorbills@odoo.com alias, Journal2 with vendorbills2@odoo.com alias) and if a vendor bill is automatically generated from `vendorbills2@odoo.com` alias then it should have `Journal2` set as it's Journal.